Amazon.IonHashDotnet/Serializer.cs

```diff
@@ -116,12 +117,17 @@ internal byte[] Digest()
             return this.HashFunction.Digest();
         }
 
-        internal void HandleFieldName(string fieldName)
+        internal void HandleFieldName(IIonHashValue ionValue, bool isInStruct)
         {
-            // the "!= null" condition allows the empty symbol to be written
-            if (fieldName != null && this.depth > 0)
+            if (this.depth > 0 && isInStruct)
             {
-                this.WriteSymbol(fieldName);
+                if (ionValue.CurrentFieldNameSymbol.Text == null &&
+                    ionValue.CurrentFieldNameSymbol.Sid == 10)
```

Comment:
SID10 is an unresolved SID which is expected to throw: https://github.com/byronlin-bq/ion-hash-dotnet/blob/master/Amazon.IonHashDotnet.Tests/IonHashWriterTest.cs#L125-L126

Discussed with Tyler and he said the GetFieldNameSymbol() api for binary reader is not expected to throw so we need to add the check in IonHash.

Reply:
SID 10 represents some user-defined symbol, and there are cases in which the reader will not be able to resolve SID10 to a string.  The same is true for SID 11, 12, 13, etc.  We need to avoid checking the value of `Sid` here; is the `Text == null` condition sufficient? (and if not, please help me understand why not?)

Reply:
`Text == null` is not sufficient because zero symbol also have null text, I updated the conditional to: 
```
                if (ionValue.CurrentFieldNameSymbol.Text == null &&
                    ionValue.CurrentFieldNameSymbol.Sid != 0)
```

Reply:
That's much more reasonable--thank you